### PR TITLE
Setting up Javascript linting. 

### PR DESCRIPTION
**Pull request by lRolwes**
**What was changed?**

JavaScript Linter added to the application. Commented code on frontend pages to add descriptions for future collaborators. Workflow added to automatically run the new JS Linter. Major linting errors fixed. 

**Why was it changed?**

These changes were made to make our code easier to navigate for future collaborators and to provide a standard for frontend code format and a means to enforce the standard.

**How was it changed?**
- ESLint was installed and a new config file added in the Frontend directory
- A yaml workfow called Frontend linting -ESLint was added to automatically execute the linting
- Header comments were added to all 3 Frontend pages. 
- Major linting errors in pages and components were fixed. 

**Screenshots that show the changes:**
image

### CODE REVIEW

File: Frontend/src/pages/SystemDetails.js [hunk 1]
```diff
@@ -8,14 +16,13 @@ import CriticalPointsTable from '../components/FunctionDetail/CriticalPointsTabl
...
 import CitationsTable from '../components/FunctionDetail/CitationsTable'
-import { useFilters } from '../context/FilterContext';
 
 function SystemDetails() {
     const [data, setData] = useState({});
     const label = useParams().label;
     useEffect(() => {
       fetchDataForCSV();
-  },[]); 
+  });
```
Comment (AbhilashKotha) [inline Frontend/src/pages/SystemDetails.js:25]: I don't think we should remove the empty array, can you check oncE?

File: Frontend/src/pages/ExploreSystems.js [hunk 2]
```diff
@@ -254,20 +263,7 @@ function ExploreSystems() {
...
-    const handlePagePerChange = (event) => {
-    // Update the state with the selected value
-	if (event.target.value == 'All'){
-	    setPagesPer(systems.length);
-	    setPagesDisplay("All");
-	} else {
-	    setPagesPer(event.target.value);
-	    setPagesDisplay(event.target.value);
-	}
-    };
```
Comment (AbhilashKotha) [inline Frontend/src/pages/ExploreSystems.js:270]: why are we removing this?

File: Frontend/src/pages/AboutPage.js [hunk 3]
```diff
@@ -15,7 +19,7 @@ function AboutPage() {
       for further analysis. This type of searchable rich data set will save researchers
       countless hours of computation as well as provide a means to identify previously
       unknown patterns and connections</p>
-      <img src="./homeImage.png" alt="image" />
```
Comment (Evan-Richmond) [inline Frontend/src/pages/AboutPage.js:18]: keep the alt here it causes an extra error and is good practice for accessibility

File: Frontend/src/pages/SystemDetails.js (hunk 1 above)
Comment (lRolwes) [reply Frontend/src/pages/SystemDetails.js:25]: I can put it back in, we just get a linting warning then.

File: Frontend/src/pages/AboutPage.js (hunk 3 above)
Comment (lRolwes) [reply Frontend/src/pages/AboutPage.js:18]: It throws a linting warning because it says the alt is already implicitly included

**Review by AbhilashKotha — APPROVED** [review]
lgtm;

**Review by Evan-Richmond — APPROVED** [review]

**Pull request merged**